### DISCUSSION TITLE
&(#(year <= 10 years)) seems to not work for me

### DISCUSSION THREAD

**Discussion by timtrace** [quodlibet/quodlibet#3852] [Q&A] [answered]
Greetings, brand-new quodlibet user here. I have MP3s with a four-digit year in the date tag, and when I run &(#(year <= 10 years)) results come back from 1996, 1977, etc. Am I doing something wrong?  Thank you!

**Comment by KarimbaYo** [comment]
Indeed, the same for me (QL 4.3.0).
I believe it's a bug...

**Comment by declension** [comment, marked as answer]
Interesting. Definitely seems like there's a bug of sorts here. 

A few notes

* Why the `&`? It seems this happens just as much with `#(year <= 10 years)`
* You can get a similar result using `#(date <= 10 years)` but it's not what you think - it's 
* I think the problem (bug) really is that `year` is not a seconds-based value, like the others, *but* it does accept the handy human shortcuts for these values (e.g. `years`)

**Reply by declension** [reply]
After checking a bit more - yes, essentially this is unsupported currently. 

`#(year < 2020)` needs to work but minutes / hours / days / years etc work by converting to seconds _first_, meaning `#(year < 10 years)` is equivalent to `#(year < 31536000)` :open_mouth: 

Something that _might_ work is a new specific tag, called `~#age`, or even maybe `#date:seconds`, though that seems clunky

**Reply by declension** [reply]
See https://github.com/quodlibet/quodlibet/issues/3927

**Comment by dosimple** [comment]
Just a side note: I feel the query language is not user friendly. Is it really comfortable to use to you?

**Reply by declension** [reply]
Yes and no given the constraints. There are two types of query that need to be detected (plain and structured) and this makes parsing pretty tricky in places otherwise

Agree that prefix notation for operators feels unintuitive though. 

For more context and ideas see my ancient rethink of this as a SQL-like language, now a Git orphan #821, and more recently discussion on #2803
